packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -1095,9 +1095,7 @@ function ChildReconciler(shouldTrackSideEffects) {
       existing.return = returnFiber;
       return existing;
     }
-    // The existing first child is not a text node so we need to create one
-    // and delete the existing ones.
-    deleteRemainingChildren(returnFiber, currentFirstChild);
+    // The existing first child is not a text node so we need to create one.
```

Comment:
If `currentFirstChild !== null && currentFirstChild.tag !== HostTest`, then yes we do need to delete the remaining children.

Reply:
Yes, I don't delete that line(which in [line 1093](https://github.com/facebook/react/pull/13745/files#diff-f8ecb7d2b88a9e6edcab34023a47fe47R1093)). I delete the line below.
Or you just mean I show adjust the comment?

Reply:
Look carefully to the code. I think @acdlite is saying we need to keep `deleteRemainingChildren(returnFiber, currentFirstChild);` in both condition? They are returning the `existing;` on line 1096, which the code would not reach.